src/utils/GenericReviver.ts

```diff
@@ -0,0 +1,37 @@
+import { loadActionIdentifier } from "../Bladeburner/utils/loadActionIdentifier";
+import { constructorsForReviver, isReviverValue } from "./JSONReviver";
+import { validateObject } from "./Validator";
+
+/**
+ * A generic "smart reviver" function.
+ * Looks for object values with a `ctor` property and a `data` property.
+ * If it finds them, and finds a matching constructor, it hands
+ * off to that `fromJSON` function, passing in the value. */
+export function Reviver(_key: string, value: unknown): any {
+  if (!isReviverValue(value)) {
+    return value;
+  }
+  const ctor = constructorsForReviver[value.ctor];
+  if (!ctor) {
+    // Known missing constructors with special handling.
+    switch (value.ctor) {
+      case "AllServersMap": // Reviver removed in v0.43.1
+      case "Industry": // No longer part of save data since v2.3.0
+      case "Employee": // Entire object removed from game in v2.2.0 (employees abstracted)
+      case "Company": // Reviver removed in v2.6.1
+      case "Faction": // Reviver removed in v2.6.1
+        console.warn(`Legacy load type ${value.ctor} converted to expected format while loading.`);
+        return value.data;
+      case "ActionIdentifier": // No longer a class as of v2.6.1
+        return loadActionIdentifier(value.data);
```

Comment:
This whole PR is good and I support it. However, I feel like the real issue is this line: A generic utility depending on a specific implementation will *always* create cycles, so you have only reduced the surface area, not eliminated them.

The cycle-removing fix would be to handle this case inside the Blade loading code, instead of here.